beps/0003-auth-architecture-evolution/README.md

```diff
@@ -120,13 +122,20 @@ export type BackstagePrincipalTypes = {
 };
 
 export interface AuthService {
-  authenticate(token: string): Promise<BackstageCredentials>;
+  authenticate(
+    token: string,
+    options?: {
+      allowLimitedAccess?: boolean;
+    },
+  ): Promise<BackstageCredentials>;
 
   isPrincipal<TType extends keyof BackstagePrincipalTypes>(
     credentials: BackstageCredentials,
     type: TType,
   ): credentials is BackstageCredentials<BackstagePrincipalTypes[TType]>;
 
+  getNoneCredentials(): Promise<BackstageCredentials<BackstageUserPrincipal>>;
```

Comment:
Just added this, it avoids a direct internal dependency between `auth`/`httpAuth`. With this in place the `HttpAuthService` implementation no longer needs to use the internal credential creators and can rely fully on the `AuthService` interface instead.